Src/LinearSolvers/MLMG/AMReX_MLMG.cpp

```diff
@@ -1132,9 +1132,11 @@ MLMG::prepareForSolve (const Vector<MultiFab*>& a_sol, const Vector<MultiFab con
     }
 
 #ifdef AMREX_USE_HYPRE
-    hypre_solver.reset();
-    hypre_bndry.reset();
-    hypre_node_solver.reset();
+    if (!preserve_hypre_solver) {
+      hypre_solver.reset();
+      hypre_bndry.reset();
+      hypre_node_solver.reset();
+    }
```

Comment:
I believe we can move this block and the petsc one below into the `else if (linop.needsUpdate())` branch above.  Then we do not need the `preserve_hypre_solver flag`.

Reply:
Thanks, I just committed this change and tested. Seems it's working as I'd expect.